pkg/app/app010_test.go

```diff
@@ -83,32 +83,36 @@ func TestApp010_AddEnvironment_empty_spec_flag(t *testing.T) {
 
 func TestApp0101_Environments(t *testing.T) {
 	withApp010Fs(t, "app010_app.yaml", func(app *App010) {
-		expected := EnvironmentSpecs{
-			"default": &EnvironmentSpec{
+		expected := EnvironmentConfigs{
+			"default": &EnvironmentConfig{
+				Name: "default",
```

Comment:
Here's what's going on - `EnvironmentConfig` objects need a `Name`, and it is derived from their key in the `EnvironmentConfigs` map. The map key and the value object's name should not be serialized separately.

Previously, this relationship was only established if you pulled out the name using the `app.GetEnvironmentConfig` accessor. This is error prone, I ran into similar related problems with registries.

What I've now down is set `EnvrionmentConfig.Name` during deserialization (see `EnvironmentConfigs.UnmarshalJSON()` - this avoids the implicit requirement to always use this accessor for an otherwise exported/public field.